packages/netlify-cms-core/src/lib/formatters.ts

```diff
@@ -166,11 +166,25 @@ export const previewUrlFormatter = (
    * URL path.
    */
   const basePath = trimEnd(baseUrl, '/');
-  const pathTemplate = collection.get('preview_path') as string;
+
+  let filePathTemplate;
```

Comment:
Can we extract getting the `pathTemplate` and `dateFieldName` to a separate function to make this one cleaner?